components/movementsensor/gpsrtkserial/gpsrtkserial.go

```diff
@@ -371,47 +390,77 @@ func (g *rtkSerial) closePort() {
 	}
 }
 
-// receiveAndWriteSerial connects to NTRIP receiver and sends correction stream to the MovementSensor through serial.
-func (g *rtkSerial) receiveAndWriteSerial() {
-	defer g.activeBackgroundWorkers.Done()
+// ntripConnection connects to NTRIP stream.
+func (g *rtkSerial) ntripConnection() error {
 	if err := g.cancelCtx.Err(); err != nil {
-		return
+		return g.err.Get()
 	}
+	g.urlMutex.Lock()
 	err := g.connect(g.ntripClient.URL, g.ntripClient.Username, g.ntripClient.Password, g.ntripClient.MaxConnectAttempts)
 	if err != nil {
 		g.err.Set(err)
-		return
+		return g.err.Get()
 	}
 
 	if !g.ntripClient.Client.IsCasterAlive() {
 		g.logger.Infof("caster %s seems to be down", g.ntripClient.URL)
 	}
 
+	g.logger.Debug("gettting source table")
+	srcTable, err := g.ntripClient.Client.ParseSourcetable()
+	if err != nil {
+		g.logger.Errorf("failed to get source table: %v", err)
+	}
+	g.logger.Debug("got sourcetable, parsing it...")
+	g.isVirtualBase, err = findLineWithMountPoint(srcTable, g.ntripClient.MountPoint)
+	if err != nil {
+		g.logger.Errorf("can't find mountpoint in source table, found err %v\n", err)
+	}
+	g.urlMutex.Unlock()
 	err = g.openPort()
 	if err != nil {
 		g.err.Set(err)
-		return
+		return g.err.Get()
```

Comment:
I'm unclear what the point of `g.err.Set(err); return g.err.Get()` is. I think the idea is that if `g.err` has a high threshold and this is the very first error, we return and pretend there was no error? but returning nil from `ntripConnection()` implies to me that we successfully connected, which seems confusing. and `newRTKSerial` always sets the threshold to 1, so I just don't get this at all. Were lines 422 and 423 supposed to be merely `return err`?

Reply:
haven't looked at the rest of the code, but we don't want to error out in the case that we're actually reading a line badly, or return from the goroutine accessing the bytes from serial and sending them to the parser continuously. If there's logic that relies on get and set error there, we should be consistent.